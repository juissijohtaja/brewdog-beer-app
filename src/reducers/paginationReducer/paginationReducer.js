const initialState = {
  page: 0,
  rowsPerPage: 10,
}

const paginationReducer = (state = initialState, action) => {
  switch (action.type) {
  case 'SET_PAGE':
    return { ...state, page: action.data }
  case 'RESET_PAGE':
    return { ...state, page: initialState.page }
  case 'SET_ROWS':
    return { ...state, rowsPerPage: action.data }
  case 'RESET_ROWS':
    return { ...state, rowsPerPage: initialState.rowsPerPage }
  default:
    return state
  }
}

export const pageChange = (value) => {
  return {
    type: 'SET_PAGE',
    data: value
  }
}

export const pageReset = () => {
  return {
    type: 'RESET_PAGE',
  }
}

export const rowsChange = (value) => {
  return {
    type: 'SET_ROWS',
    data: value
  }
}

export const rowsReset = () => {
  return {
    type: 'RESET_ROWS',
  }
}

export default paginationReducer