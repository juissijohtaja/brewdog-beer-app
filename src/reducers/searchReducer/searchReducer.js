const searchReducer = (state = '', action) => {
  switch (action.type) {
  case 'SET_KEYWORD':
    return action.keyword
  case 'RESET_KEYWORD':
    return ''
  default:
    return state
  }
}

export const keywordChange = keyword => ({ type: 'SET_KEYWORD', keyword })
export const keywordReset = () => ({ type: 'RESET_KEYWORD' })

export default searchReducer