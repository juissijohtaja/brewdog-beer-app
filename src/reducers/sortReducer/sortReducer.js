const sortReducer = (state = true, action) => {
  switch (action.type) {
  case 'SET_SORT':
    return !state
  case 'RESET_SORT':
    return true
  default:
    return state
  }
}

export const sortChange = () => ({ type: 'SET_SORT' })
export const sortReset = () => ({ type: 'RESET_SORT' })

export default sortReducer