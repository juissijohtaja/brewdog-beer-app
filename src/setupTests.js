import '@testing-library/jest-dom'

const initialStoreState = {
  beers: [],
  keyword: '',
  pagination: {
    page: 0,
    rowsPerPage: 10
  },
  sortDesc: true
}

export const changedStoreState = {
  beers: [{
    id: 999,
    beer: 1111,
    description: 'Lorem ipsum dolor sit amet',
    startDate: '1999-12-31T00:00:00+00:00',
    category: 'Test',
    responsible: 'John Doe',
    savingsAmount: 1234.56,
    currency: 'EUR',
    complexity: 'Simple'
  },
  {
    id: 9998,
    beer: '',
    description: '',
    startDate: '',
    category: '',
    responsible: '',
    savingsAmount: '',
    currency: '',
    complexity: ''
  }],
  keyword: 'lorem',
  pagination: {
    page: 0,
    rowsPerPage: -1
  },
  sortDesc: false
}

export default initialStoreState