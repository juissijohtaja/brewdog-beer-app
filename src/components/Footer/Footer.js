import React from 'react'
import { useDispatch } from 'react-redux'
import { makeStyles } from '@material-ui/core/styles'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import RotateLeftIcon from '@material-ui/icons/RotateLeft'
import { grey } from '@material-ui/core/colors'
import { keywordReset } from '../../reducers/searchReducer/searchReducer'
import { pageReset, rowsReset } from '../../reducers/paginationReducer/paginationReducer'
import { sortReset } from '../../reducers/sortReducer/sortReducer'

const useStyles = makeStyles(theme => ({
  footerRoot: {
    background: grey[900],
    [theme.breakpoints.down('sm')]: {
      width: '100%',
      position: 'fixed',
      bottom: 0,
    },
  },
  actionRoot: {
    color: grey[600],
    '&$actionSelected': {
      color: grey[50]
    }
  },
  actionSelected: {
    color: grey[50]
  }
}))

const Footer = () => {
  const dispatch = useDispatch()
  const classes = useStyles()
  const actionClasses = {
    root: classes.actionRoot,
    selected: classes.actionSelected,
  }
  const [value, setValue] = React.useState(0)

  const handleReset = () => {
    dispatch(keywordReset())
    dispatch(pageReset())
    dispatch(rowsReset())
    dispatch(sortReset())
  }

  return (
    <BottomNavigation
      value={value}
      onChange={(event, newValue) => {
        setValue(newValue)
      }}
      showLabels
      classes={{ root: classes.footerRoot }}
      data-testid='footer'
    >
      <BottomNavigationAction
        onClick={handleReset}
        classes={actionClasses}
        label='Reset'
        icon={<RotateLeftIcon />}
        aria-label='reset view'
      />
    </BottomNavigation>
  )
}

export default Footer