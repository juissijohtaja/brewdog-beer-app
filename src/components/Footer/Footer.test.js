import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { initialStoreState } from '../../setupTests'
import Footer from './Footer'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { keywordReset } from '../../reducers/searchReducer/searchReducer'
import { pageReset, rowsReset } from '../../reducers/paginationReducer/paginationReducer'
import { sortReset } from '../../reducers/sortReducer/sortReducer'

describe('Footer Component', () => {
  let store, component
  const mockStore = configureStore([])

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()

    component = renderer.create(
      <Provider store={store}>
        <Footer />
      </Provider>
    )
  })

  it('should render BottomNavigation and Reset button', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should dispatch 4 actions on button click', () => {
    renderer.act(() => {
      component.root.findByType('button').props.onClick()
    })

    expect(store.dispatch).toHaveBeenCalledTimes(4)
    expect(store.dispatch).toHaveBeenCalledWith(
      sortReset()
    )
    expect(store.dispatch).toHaveBeenCalledWith(
      pageReset()
    )
    expect(store.dispatch).toHaveBeenCalledWith(
      rowsReset()
    )
    expect(store.dispatch).toHaveBeenCalledWith(
      keywordReset()
    )
  })
})