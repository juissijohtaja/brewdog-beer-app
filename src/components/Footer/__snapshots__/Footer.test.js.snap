// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render BottomNavigation and Reset button 1`] = `
<div
  className="MuiBottomNavigation-root makeStyles-footerRoot-1"
  data-testid="footer"
>
  <button
    aria-label="reset view"
    className="MuiButtonBase-root MuiBottomNavigationAction-root makeStyles-actionRoot-2 Mui-selected makeStyles-actionSelected-3"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.11 8.53L5.7 7.11C4.8 8.27 4.24 9.61 4.07 11h2.02c.14-.87.49-1.72 1.02-2.47zM6.09 13H4.07c.17 1.39.72 2.73 1.62 3.89l1.41-1.42c-.52-.75-.87-1.59-1.01-2.47zm1.01 5.32c1.16.9 2.51 1.44 3.9 1.61V17.9c-.87-.15-1.71-.49-2.46-1.03L7.1 18.32zM13 4.07V1L8.45 5.55 13 10V6.09c2.84.48 5 2.94 5 5.91s-2.16 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93s-3.05-7.44-7-7.93z"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label Mui-selected makeStyles-actionSelected-3"
      >
        Reset
      </span>
    </span>
  </button>
</div>
`;
